File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    let check = new Array(n).fill(false);

    
    for(let i=0; i<n; i++){
        if(check[i]===false){
            dfs(n,computers,i,check);
            answer++;
        }
    }
    
    return answer;
}

function dfs(n,computers,i,check){
    check[i]=true;
    
    for(let k=0; k<n; k++){
        if(i!==k && computers[i][k]===1 && check[k]===false) dfs(n,computers,k,check);
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    memo = {}
    visited = [False]*len(computers)
    
    for i,c in enumerate(computers):
        memo[i] = c
    
    def dfs(computer):
        if visited[computer] == True:
            return
        
        visited[computer] = True
        
        for i,v in enumerate(memo[computer]):
            if visited[i] == False and v == 1:
                dfs(i)
        
    for i in range(len(computers)):
        if visited[i] == False:
            dfs(i)
            answer+=1
    
    
    return answer